tests/syft/grid/services/signaling_service_test.py

```diff
@@ -1,48 +1,226 @@
+        address=target,
+        payload="SDP",
+        host_metadata=bob_vm.get_metadata_for_client(),
+        target_peer=target,
+        host_peer=bob_vm.address,
+    )
 
     blob = msg.serialize()
     msg2 = sy.deserialize(blob=blob)
 
     assert msg.id == msg2.id
-    assert msg.address == msg2.address
+    assert msg.address == target
     assert msg.payload == msg2.payload
-    assert msg2.payload == "Test Payload"
+    assert msg2.payload == "SDP"
+    assert msg2.host_metadata == bob_vm.get_metadata_for_client()
     assert msg == msg2
 
 
-def test_signaling_service() -> None:
+def test_signaling_answer_pull_request_message_serde() -> None:
     bob_vm = sy.VirtualMachine(name="Bob")
-    bob_vm.immediate_services_without_reply.append(SignalingService)
-    bob_vm._register_services()  # re-register all services including SignalingService
-    bob_vm_client = bob_vm.get_root_client()
+    target = Address(name="Alice")
+
+    msg = AnswerPullRequestMessage(
+        address=target,
+        target_peer=target,
+        host_peer=bob_vm.address,
+        reply_to=bob_vm.address,
+    )
+
+    blob = msg.serialize()
+    msg2 = sy.deserialize(blob=blob)
+
+    assert msg.id == msg2.id
+    assert msg.address == target
+    assert msg == msg2
+
+
+def test_signaling_offer_pull_request_message_serde() -> None:
+    bob_vm = sy.VirtualMachine(name="Bob")
+    target = Address(name="Alice")
+
+    msg = OfferPullRequestMessage(
+        address=target,
+        target_peer=target,
+        host_peer=bob_vm.address,
+        reply_to=bob_vm.address,
+    )
+
+    blob = msg.serialize()
+    msg2 = sy.deserialize(blob=blob)
+
+    assert msg.id == msg2.id
+    assert msg.address == target
+    assert msg == msg2
+
+
+def test_push_offer_signaling_service() -> None:
+    om_network, bob_vm, alice_vm = get_preset_nodes()
+    om_network_client = om_network.get_root_client()
+
+    offer_msg = SignalingOfferMessage(
+        address=om_network.address,
+        payload="SDP",
+        host_metadata=alice_vm.get_metadata_for_client(),
+        target_peer=alice_vm.address,
+        host_peer=bob_vm.address,
+    )
+    om_network_client.send_immediate_msg_without_reply(msg=offer_msg)
+
+    assert om_network.in_memory_client_registry.pop(offer_msg.id) == offer_msg
+
 
-    bob_vm_client.send_immediate_msg_without_reply(
-        msg=SignalingAnswerMessage(address=bob_vm.address, payload="Test Payload")
+def test_push_answer_signaling_service() -> None:
+    om_network, bob_vm, alice_vm = get_preset_nodes()
+    om_network_client = om_network.get_root_client()
+
+    ans_msg = SignalingAnswerMessage(
+        address=om_network.address,
+        payload="SDP",
+        host_metadata=alice_vm.get_metadata_for_client(),
+        target_peer=alice_vm.address,
+        host_peer=bob_vm.address,
+    )
+    om_network_client.send_immediate_msg_without_reply(msg=ans_msg)
+
+    assert om_network.in_memory_client_registry.pop(ans_msg.id) == ans_msg
```

Comment:
The ideal structure to save these messages should be the `node.store`, but since `SyftMessages` are not currently storable, I chose to save them in a secondary structure (which should be removed in future versions, forcing us to return to this issue). In another PR, I intend to make `SyftMessages` (at least those related to `SignalingService`) into storable objects. We need to store them in order to provide scalability (adding Redis cache, etc...)

Reply:
Oh interesting - we should probably make SyftMessages storable in the database...

Reply:
Hmm - it makes me nervous to merge this kind of temporary fix because I fear we'll never come back to it.

Reply:
I'd rather create a global list attached to the worker in memory (which will stick out like a sore thumb) than store it in a list that isn't expecting it. Can we do that instead?

Reply:
Yup!

Reply:
Fixed! 25da489

Reply:
Looks resolved as described.